### ISSUE TITLE
Specs for deprecated methods

### ISSUE THREAD

**Issue by andrykonchin** [ruby/spec#780]
There are some deprecated/obsolete methods like `File.exists?`, `Thread#safe_level` etc in the core library. 

Looks like they aren't covered with tests there and moreover some existing specs were deleted recently - `File.exists?`, `Dir.exists? and `FileTest.exists?` ([commit](https://github.com/ruby/spec/commit/79e4a3d7f5a2e81cefa824794ad82f704121dc7c)).

So I am wondering whether it is a well-considered decision or it's unintentional situation. I believe that as far as these methods are still available in MRI they should be a part of the specification.

**Comment by eregon** [comment]
I agree, the specs should remain as long as there is a non-EOL version that supports it.

We just need to find a way to silence those deprecation warnings for the deprecated but ideally not for the non-deprecated methods.
